src/test/app/DepositAuth_test.cpp

```diff
@@ -724,14 +792,506 @@ struct DepositPreauth_test : public beast::unit_test::suite
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                // AuthorizeCredentials is larger than 8 elements
+                Account const a("a"), b("b"), c("c"), d("d"), e("e"), f("f"),
+                    g("g"), h("h"), i("i");
+                auto const& z = credType;
+                auto jv = deposit::authCredentials(
+                    bob,
+                    {{a, z},
+                     {b, z},
+                     {c, z},
+                     {d, z},
+                     {e, z},
+                     {f, z},
+                     {g, z},
+                     {h, z},
+                     {i, z}});
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                // Can't create with non-existing issuer
+                Account const rick{"rick"};
+                auto jv = deposit::authCredentials(bob, {{rick, credType}});
+                env(jv, ter(tecNO_ISSUER));
+                env.close();
+            }
+
+            {
+                // not enough resevre
+                Account const john{"john"};
+                env.fund(env.current()->fees().accountReserve(0), john);
+                auto jv = deposit::authCredentials(john, {{issuer, credType}});
+                env(jv, ter(tecINSUFFICIENT_RESERVE));
+            }
+
+            {
+                // NO deposit object exists
+                env(deposit::unauthCredentials(bob, {{issuer, credType}}),
+                    ter(tecNO_ENTRY));
+            }
+
+            // Create DepositPreauth object
+            {
+                env(deposit::authCredentials(bob, {{issuer, credType}}));
+                env.close();
+
+                auto const jDP =
+                    ledgerEntryDepositPreauth(env, bob, {{issuer, credType}});
+                BEAST_EXPECT(
+                    jDP.isObject() && jDP.isMember(jss::result) &&
+                    !jDP[jss::result].isMember(jss::error) &&
+                    jDP[jss::result].isMember(jss::node) &&
+                    jDP[jss::result][jss::node].isMember("LedgerEntryType") &&
+                    jDP[jss::result][jss::node]["LedgerEntryType"] ==
+                        jss::DepositPreauth);
+
+                // can't create duplicate
+                env(deposit::authCredentials(bob, {{issuer, credType}}),
+                    ter(tecDUPLICATE));
+            }
+
+            // Delete DepositPreauth object
+            {
+                env(deposit::unauthCredentials(bob, {{issuer, credType}}));
+                env.close();
+                auto const jDP =
+                    ledgerEntryDepositPreauth(env, bob, {{issuer, credType}});
+                BEAST_EXPECT(
+                    jDP.isObject() && jDP.isMember(jss::result) &&
+                    jDP[jss::result].isMember(jss::error) &&
+                    jDP[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+
+        {
```

Comment:
Can these tests be moved up to where the other payment tests are? It's a bit confusing that they're separated.

Reply:
divided to  testCredentialsPayment();  testCredentialsCreation();